### DISCUSSION TITLE
Welcome to ngx-capture Discussions!

### DISCUSSION THREAD

**Discussion by Wanchai** [Wanchai/ngx-capture#11] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by arashetm** [comment]
How can I use `getImage` function in angular 9.1.6?!
It returns `Promise<string>` that we can't _pipe_ or _subscribe_ it,
also, it has an error:

`DOMException: Failed to set the 'adoptedStyleSheets' property on 'ShadowRoot': Sharing constructed stylesheets in multiple documents is not allowed`

I use the function in this way:

`this.captureService.getImage(this.screen.nativeElement, true).then((response) => {
      console.log(response);
    })`

**Reply by Wanchai** [reply]
A promise can ber transformed into an observable with the 'from' function of Rxjs.

The error you showing me do not seem related to ngx-capture. With a little more context, I could probably help you.

**Comment by wynnteo** [comment]
How can we capture an image with a source of a different origin?

**Reply by Wanchai** [reply]
You can't.
It's a security feature from browsers and you can't go over it, sorry.

**Comment by AjCreations195** [comment]
How can I capture screenshot of any website by typing url?

**Reply by Wanchai** [reply]
The purpose is to capture your own website.

**Comment by AjCreations195** [comment]
Can I use this method to capture screenshot of website  using URL  in Angular?

**Reply by Wanchai** [reply]
The purpose is to capture your own website.

**Comment by karthi2912002** [comment]
Is this ngx-capture can able to take screenshot with the Grid component

**Reply by Wanchai** [reply]
I need more details. Which grid component?

**Reply by karthi2912002** [reply]
I am using Ag-grid material for grids when i try to select a particular part of my app it throw error like unsupported "color". the error was throwed by the html2canvas, I tried different approach like to neutralize the color component style.
